lib/sinatra/base.rb

```diff
@@ -1005,7 +1005,8 @@ def process_route(pattern, keys, conditions, block = nil, values = [])
       values += match.captures.map! { |v| force_encoding URI_INSTANCE.unescape(v) if v }
 
       if values.any?
-        original, @params = params, params.merge('splat' => [], 'captures' => values)
+        original = @params
+        @params = @params.merge('splat' => [], 'captures' => values)  # makes a copy of @params
```

Comment:
Right now I am solving by adding the line
<pre><code>return @params unless @params['splat']</code></pre>
to the top of my overriding method.